library/proc_macro/src/lib.rs

```diff
@@ -1156,6 +1162,28 @@ impl Literal {
     }
 }
 
+/// Parse a single literal from its stringified representation.
+///
+/// In order to parse successfully, the input string must not contain anything
+/// but the literal token. Specifically, it must not contain whitespace or
+/// comments in addition to the literal.
+///
+/// The resulting literal token will have a `Span::call_site()` span.
+///
+/// NOTE: some errors may cause panics instead of returning `LexError`. We
+/// reserve the right to change these errors into `LexError`s later.
+#[stable(feature = "proc_macro_literal_parse", since = "1.53.0")]
+impl FromStr for Literal {
+    type Err = LexError;
```

Comment:
This error is displayed as *"cannot parse string into token stream"*. Should we make that text a bit more generic, or alternatively use a different error type?

Reply:
I think this is fine as is. LexError is useless in practice because the parser in proc-macro will panic instead of ever returning it, so nothing in the ecosystem relies on having a meaningful LexError. (Actually it's worse -- sometimes the parser will return `Ok` even for input strings that are invalid tokens.)

Whenever the parser is fixed to be able to reliably return LexError instead of panicking or pretending like it worked, we'll likely consider adding some internal enum into LexError to provide more info on what happened. This new code would take advantage of that at the same time.